x-pack/plugins/ml/public/application/data_frame_analytics/pages/analytics_exploration/components/exploration_query_bar/exploration_query_bar.tsx

```diff
@@ -59,6 +60,31 @@ export const ExplorationQueryBar: FC<ExplorationQueryBarProps> = ({
 
   const searchChangeHandler = (q: Query) => setSearchInput(q);
 
+  const regex = useMemo(() => new RegExp(`${filters?.columnId}\s?:\s?(true|false)`, 'g'), []);
+
+  /**
+   * If a filter option is active in the url set the corresponding options
+   * button to selected mode
+   */
+  useEffect(() => {
+    if (filters !== undefined) {
+      const match: string[] | null = query.query.match(regex);
+      let filterKeyInEffect: string | undefined;
+
+      if (match !== null && match[0].includes('true')) {
+        // set { training: true }
```

Comment:
Can you explain how the code knows this is about the `training` attribute? Or is this just an example and it's applicable to attributes in general?

Reply:
Good question - the filter `key` property (filters are passed in as props) shows what field value the filter option corresponds to to be 'selected'

So for example the filter's `columnId` is `ml.is_training'`and the key for Training is `training: true` - so `ml.is_training` value would be`true`. The key for `Testing` is `testing: false` meaning `ml.is_training` value should be `false` when this is selected.

Reply:
Oh so "filter" refers to what's available as the buttons and not something derived from the free text query is that correct, that's why it can only be `training`?

Reply:
Yes, "filters" are what's available as the buttons and is passed into this component from the parent component.
`training` is just an example since that's the only current value is but it would be whatever is passed in as a filter.
 
🤔 Maybe this comment is more confusing than clarifying. Would it help to replace it with something like `<filter_key_value: true`> ? Or remove it entirely and add a comment for an example filter up by the filter type?

Reply:
Think it's fine to leave as is - just maybe update the comment above before the `useEffect` to emphasize this is about "restoring state from the URL once on load" or similar - I missed that and only now noticed the `[]` to make it trigger only once. 👍

Reply:
Updated in a349738dc5dddfb51907e5d04aa22126df55ff9e